tests/unit/components/cylc/gantt/filter.spec.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  matchTasks,
+  platformOptions
+} from '@/components/cylc/gantt/filter'
+
+describe('matchTasks', () => {
+  const tasksIn = {
+    task_1: [{
+      task_name: 'task_1',
+      task_platform: 'test_platform',
+      some_other_key: 'some_other_test_key'
+    }],
+    task_2: [{
+      task_name: 'task_2',
+      task_platform: 'test_platform_2',
+      some_other_key: 'some_other_test_key'
+    }],
+    task_3: [{
+      task_name: 'task_3',
+      task_platform: 'test_platform_3',
+      some_other_key: 'some_other_test_key'
+    }]
+  }
+  const tasksOut = {
+    task_1: [{
+      task_name: 'task_1',
+      task_platform: 'test_platform',
+      some_other_key: 'some_other_test_key'
+    }],
+    task_2: [{
+      task_name: 'task_2',
+      task_platform: 'test_platform_2',
+      some_other_key: 'some_other_test_key'
+    }],
+  }
+
+  it('should filter out the tasks that do not match', () => {
+    const filter = {
+      name: ['task_1', 'task_2'],
+      platformOption: -1,
```

Comment:
Ought to have another test that tests filtering by platform as well

Reply:
Done